src/main/java/controller/EditRuleController.java

```diff
@@ -78,40 +89,88 @@ private void onSaveRule() {
 			return;
 		}
 
-		String newRule = getJavascriptString(rawRuleConditions);
-
-		// gets the rules list from the main controller
-		ArrayList<CodeQualityRule> rulesList = mainC.getRulesList();
-
-		if (!rulesList.contains(rule)) {
-			rule = new CodeQualityRule(newName, newRule, false, editRulePopup.isAdvancedMode());
-		} else {
-			rulesList.remove(rule);
-
-			if (rule.isDefault()) {
-				rule = new CodeQualityRule(rule.getName(), newRule, true, rule.isAdvanced());
-			} else {
-				rule = new CodeQualityRule(newName, newRule, false, false);
-			}
+		String newRule = getJavascriptIfStatementString(rawRuleConditions);
 
+		try {
+			// Runs pre validation to try to catch some errors
+			preValidateJavascriptCode(newRule);
+		} catch (ScriptException e) {
+			editRulePopup.showMessage("The rule provided has an invalid format. Cannot save it.");
+			return;
+		}
+		
+		if (!validadeDefaultRuleThresholds(rule.getName(), newRule)) {
+			editRulePopup.showMessage("This is a Default Rule. As such, only the thresholds \n can be edited. And values must be positive.");
+			return;
 		}
 
+
 		// updates the list with the new (validated) rule
 		// updates the list for the main controller
-		rulesList.add(rule);
+		if (!rule.isDefault()) {
+			rule.setName(newName);
+		}
+		rule.setRule(newRule);
```

Comment:
no validation for rule being default? where's the pretty regex from yesterday?

Reply:
I found the regex but am very confused as to how you apply it 1) only to default rules and 2) how it avoids us getting here and doing whatever we want to any rule

Reply:
I'll comment in the relevant blocks of code about it, so it is clearer to get.